core/finality-grandpa/src/environment.rs

```diff
@@ -80,34 +82,45 @@ const NUM_LAST_COMPLETED_ROUNDS: usize = 2;
 
 impl<Block: BlockT> Encode for CompletedRounds<Block> {
 	fn encode(&self) -> Vec<u8> {
-		Vec::from_iter(&self.inner).encode()
+		let v = Vec::from_iter(&self.rounds);
+		(&v, &self.set_id, &self.voters).encode()
 	}
 }
 
 impl<Block: BlockT> Decode for CompletedRounds<Block> {
 	fn decode<I: parity_codec::Input>(value: &mut I) -> Option<Self> {
-		Vec::<CompletedRound<Block>>::decode(value)
-			.map(|completed_rounds| CompletedRounds {
-				inner: completed_rounds.into(),
+		<(Vec<CompletedRound<Block>>, u64, Vec<AuthorityId>)>::decode(value)
```

Comment:
We should have a migration for this in `aux_schema`, I don't think we need to create a new version. We can try decoding `(Vec<CompletedRound<Block>>, u64, Vec<AuthorityId>)` and if it fails try `Vec<CompletedRound<Block>>` and use the values from the decoded `AuthoritySet`.